execute/plugin.go

```diff
@@ -562,18 +589,68 @@ func (p *Plugin) Reports(seqNr uint64, outcome ocr3types.Outcome) ([]ocr3types.R
 func (p *Plugin) ShouldAcceptAttestedReport(
 	ctx context.Context, u uint64, r ocr3types.ReportWithInfo[[]byte],
 ) (bool, error) {
-	panic("implement me")
+	decodedReport, err := p.reportCodec.Decode(ctx, r.Report)
+	if err != nil {
+		return false, fmt.Errorf("decode commit plugin report: %w", err)
+	}
+
+	if len(decodedReport.ChainReports) == 0 {
+		p.lggr.Infow("skipping empty report")
+		return false, nil
+	}
+	return true, nil
 }
 
 func (p *Plugin) ShouldTransmitAcceptedReport(
 	ctx context.Context, u uint64, r ocr3types.ReportWithInfo[[]byte],
 ) (bool, error) {
-	panic("implement me")
+	isWriter, err := p.supportsDestChain()
+	if err != nil {
+		return false, fmt.Errorf("can't know if it's a writer: %w", err)
+	}
+	if !isWriter {
+		p.lggr.Debugw("not a writer, skipping report transmission")
+		return false, nil
+	}
+
+	decodedReport, err := p.reportCodec.Decode(ctx, r.Report)
+	if err != nil {
+		return false, fmt.Errorf("decode commit plugin report: %w", err)
+	}
+
+	// TODO: Final validation?
+
+	p.lggr.Debugw("transmitting report",
+		"reports", len(decodedReport.ChainReports),
+	)
+	return true, nil
 }
 
 func (p *Plugin) Close() error {
 	panic("implement me")
 }
 
+func (p *Plugin) supportedChains(id commontypes.OracleID) (mapset.Set[cciptypes.ChainSelector], error) {
+	p2pID, exists := p.oracleIDToP2pID[id]
+	if !exists {
+		return nil, fmt.Errorf("oracle ID %d not found in oracleIDToP2pID", p.reportingCfg.OracleID)
+	}
+	supportedChains, err := p.homeChain.GetSupportedChainsForPeer(p2pID)
+	if err != nil {
+		p.lggr.Warnw("error getting supported chains", err)
+		return mapset.NewSet[cciptypes.ChainSelector](), fmt.Errorf("error getting supported chains: %w", err)
+	}
+
+	return supportedChains, nil
+}
+
+func (p *Plugin) supportsDestChain() (bool, error) {
+	destChainConfig, err := p.homeChain.GetChainConfig(p.cfg.DestChain)
+	if err != nil {
+		return false, fmt.Errorf("get chain config: %w", err)
+	}
+	return destChainConfig.SupportedNodes.Contains(p.oracleIDToP2pID[p.reportingCfg.OracleID]), nil
```

Comment:
Should we also check that `p.oracleIDToP2pID[p.reportingCfg.OracleID]` exists? Like it was done in `supportedChains`?

Reply:
maybe better to do:

```
chains, err := p.supportedChains(p.reportingCfg.OracleID)
if err != nil {
	return false, fmt.Errorf("error getting supported chains: %w", err)
}
return chains.Contains(p.cfg.DestChain), nil
```